src/test/java/org/folio/linked/data/integration/kafka/sender/search/AuthorityCreateMessageSenderTest.java

```diff
@@ -0,0 +1,123 @@
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.context.ApplicationEventPublisher;
+
+@UnitTest
+@ExtendWith(MockitoExtension.class)
+public class AuthorityCreateMessageSenderTest {
+
+  @InjectMocks
+  private AuthorityCreateMessageSender producer;
+  @Mock
+  private ApplicationEventPublisher eventPublisher;
+  @Mock
+  private AuthoritySearchMessageMapper authoritySearchMessageMapper;
+  @Mock
+  private FolioMessageProducer<ResourceIndexEvent> resourceIndexEventMessageProducer;
+
+  @Test
+  public void sendAuthorityCreated_shouldNotSendMessageAndPublishEvent_ifAuthorityAlreadyIndexed() {
+    // given
+    var resource = new Resource().addTypes(PERSON).setId(randomLong());
+    resource.setIndexDate(new Date());
+
+    // when
+    producer.produce(resource);
+
+    // then
+    verifyNoInteractions(eventPublisher, resourceIndexEventMessageProducer);
+  }
+
+  @Test
+  public void sendAuthorityCreated_shouldSendMessageAndPublishEvent_ifAuthorityNotIndexed() {
+    // given
+    var resource = new Resource().addTypes(PERSON).setId(randomLong());
+    var index = new LinkedDataAuthority().id(String.valueOf(resource.getId()));
+    when(authoritySearchMessageMapper.toIndex(resource, CREATE))
+      .thenReturn(Optional.of(index));
+
+    // when
+    producer.produce(resource);
+
+    // then
+    verifyMessageAndEvent(index);
+  }
+
+  @Test
+  public void sendAuthorityCreated_shouldNotSendMessageAndPublishEvent_ifAuthorityAlreadyIndexedAndLinkedToWork() {
+    // given
+    var work = new Resource().addTypes(WORK).setId(randomLong());
+    var authority = new Resource().addTypes(PERSON).setId(randomLong());
+    authority.setIndexDate(new Date());
+    work.getOutgoingEdges().add(new ResourceEdge(work, authority, PredicateDictionary.NULL));
+
+    // when
+    producer.produce(work);
+
+    // then
+    verifyNoInteractions(eventPublisher, resourceIndexEventMessageProducer);
+  }
+
+  @Test
+  public void sendAuthorityCreated_shouldSendMessageAndPublishEvent_ifAuthorityLinkedToWork() {
+    // given
+    var work = new Resource().addTypes(WORK).setId(randomLong());
+    var authority = new Resource().addTypes(PERSON).setId(randomLong());
+    work.getOutgoingEdges().add(new ResourceEdge(work, authority, PredicateDictionary.NULL));
+
+    var index = new LinkedDataAuthority().id(String.valueOf(authority.getId()));
+    when(authoritySearchMessageMapper.toIndex(authority, CREATE))
+      .thenReturn(Optional.of(index));
+
+    // when
+    producer.produce(work);
+
+    // then
+    verifyMessageAndEvent(index);
+  }
+
+  private void verifyMessageAndEvent(LinkedDataAuthority index) {
+    var messageCaptor = ArgumentCaptor.forClass(List.class);
+    verify(resourceIndexEventMessageProducer)
+      .sendMessages(messageCaptor.capture());
+    List<ResourceIndexEvent> messages = messageCaptor.getValue();
```

Comment:
**var** messages

Reply:
need to define type explicitly
<img width="200" alt="image" src="https://github.com/user-attachments/assets/a733b2fe-5606-441f-826d-e962c2f2b33b">